javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -133,12 +135,13 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
+        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
         Supplier<? extends T> tmp = supplier;
```

Comment:
I strongly think we should base speedups on JMH benchmarks, our intuition is most often wrong about these ...